### ISSUE TITLE
[GO] If a path argument is named 'path', generated API method will throw a build time error

### ISSUE THREAD

**Issue by ivucica** [swagger-api/swagger-codegen#3566]
<!--
Please follow the issue template below for bug reports and feature requests.
Also please indicate in the issue title which language/library is concerned. Eg:  [JAVA] Bug generating foo with bar 
-->

##### Description

<!-- describe what is the issue and why this is a problem for you. -->
Function arguments are named after path parameters in the API spec. If a path parameter is named `path`, this generates a build time error as Go does not like reusing the name to 'create' a 'new' variable as long as there is a preexisting variable with the same name in the scope.

##### Swagger-codegen version

<!-- which version of swagger-codegen are you using, is it a regression? -->
7a245e31894c30a050e59cffefb97739005c0013

##### Swagger declaration file content or url

<!-- if it is a bug, a json or yaml that produces it.
    If you post the code inline, please wrap it with
    ```yaml
    (here your code)
    ```
    (for YAML code) or
    ```json
    (here your code)
    ```
    (for JSON code), so it becomes more readable. If it is longer than about ten lines,
    please create a Gist (https://gist.github.com) or upload it somewhere else and
    link it here.
  -->
Full file available at https://bitbucket.org/api/swagger.json (it may change, but that's out of my control)

Also https://bitbucket.org/ivucica/calendarify/src/f1297e3411fcb52e830bd02dd0108e85eed39488/vendor/bitbucket.org/api/swagger.json

##### Command line used for generation

<!-- including the language, libraries and various options -->

See https://bitbucket.org/ivucica/calendarify/src/f1297e3411fcb52e830bd02dd0108e85eed39488/vendor/bitbucket.org/api/BUILD

```python
genrule(
    name = "rest_client_go_gen",
    srcs = [":swagger.json"],
    outs = SWAGGER_GENERATED,
    tools = [
        "//vendor:swagger-codegen-cli",
        "//vendor:swagger-codegen-cli_deploy.jar",
    ],
    cmd = "\n".join([
        "mkdir -p \"$$(dirname $(location :configuration.go))\"",
        "$(location //vendor:swagger-codegen-cli) --singlejar generate " +
        " -i $(location :swagger.json)" +
        " -l go" +
        " -o \"$$(dirname $(location :configuration.go))\"",
    ]),
)
```

Versions of dependencies: https://bitbucket.org/ivucica/calendarify/src/f1297e3411fcb52e830bd02dd0108e85eed39488/WORKSPACE

Versions of swagger projects: https://bitbucket.org/ivucica/calendarify/src/f1297e3411fcb52e830bd02dd0108e85eed39488/vendor/github.com/swagger-api/

- swagger-codegen [7a245e31894c]
- swagger-core [c7c9a4c641dc]
- swagger-parser [0e2fac8d6e46]
- swagger-ui [44149467753f]


##### Steps to reproduce

<!-- unambiguous set of steps to reproduce the bug.-->
Bitbucket's API contains the following definition (transformed from JSON into YAML by the Swagger editor):

```yaml
paths:
...
  '/repositories/{username}/{repo_slug}/issues/{issue_id}/attachments/{path}':
    delete:
      ...
    parameters:
      - required: true
        type: string
        name: username
        in: path
      - required: true
        type: string
        name: path
        in: path
      - required: true
        type: string
        name: issue_id
        in: path
      - required: true
        type: string
        name: repo_slug
        in: path
    ...
```

Generated code looks like:

```go
func (a IssuetrackerApi) RepositoriesUsernameRepoSlugIssuesIssueIdAttachmentsPathDelete(username string, path string, issueId string, repoSlug string) (*APIResponse, error) {                                                                                         
                                                                                                                                   
        var httpMethod = "Delete"                                                                                                   
        // create path and map variables                                                                                            
        path := a.Configuration.BasePath + "/repositories/{username}/{repo_slug}/issues/{issue_id}/attachments/{path}"              
...
```
(note the 'path' is a preexisting variable, used as an argument.)

This results in a build error in Go such as:      

```
badc0de.net/pkg/calendarify/bazel-out/local-fastbuild/genfiles/vendor/bitbucket.org/api/issuetracker_api.go:342: no new variables on left side of :=

```

##### Related issues

<!-- has a similar issue been reported before? -->
I am unaware of any.

##### Suggest a Fix

<!-- if you can't fix the bug yourself, perhaps you can point to what might be
  causing the problem (line of code or commit) -->
If a method argument would be named `path`, change it to path_arg, or some other value that would not be used in the generated code.

**Comment by wing328** [comment]
@ivucica thanks for reporting the issue. I think we'll do something similar to other API clients by prefixing local variable with localVar{name} (e.g. name = Path) and put localVar{name} in the reserved keyword list.

**Comment by wing328** [comment]
Resolved via #3570.

Please pull the latest master to give it a try.

**Comment by ivucica** [comment]
This is verified to fix the issue.

To make the API generated from Bitbucket's spec build, only bug #3565 and PR #3573 remain.
